kit/Kit.cpp

```diff
@@ -2681,7 +2682,9 @@ int KitSocketPoll::kitPoll(int timeoutMicroS)
         do
         {
             int realTimeout = timeoutMicroS;
-            if (_document && _document->hasQueueItems() && _document->processInputEnabled())
+            if (_document &&
+                (_document->hasCallbacks() ||
+                 (_document->hasQueueItems() && _document->processInputEnabled())))
```

Comment:
Reassure me that this  (a || b && c) should not be (a || b) && c, we want this as it is written?

Reply:
Yes - it is correct; I can split it into two happier named booleans if that helps =)

Reply:
Split to its own function & commented it :-)

Reply:
great :-)